client/src/components/subscription/createSubscription.js

```diff
@@ -55,11 +69,88 @@ class CreateSubscription extends React.Component {
     toast.success('Successfully created subscription!');
 
     this.handleSubscriptions(response);
-    this.setState({ name: '', nickname: '', dueDate: todaysDate, reminderDays: 0, amount: 0 });
+    this.setState({ name: '', nickname: '', reminderDays: 0, amount: 0, frequency: '', occurence: 0, days: []});
+  }
+
+  handleDays(days) {
+    this.setState({ days });
+  }
+
+  parseDueDate() {
+    const { frequency, days } = this.state;
+    let parsedDay;
+    switch (frequency) {
+      case 'yearly':
+        parsedDay = days.map((day) => day.toISOString());
+        break;
+      case 'monthly':
+        parsedDay = days.map((day) => day.toISOString());
+        break;
+      case 'weekly':
+        parsedDay = days.map((day) => day.getDay());
+        break;
+      case 'daily':
+        parsedDay = null;
+        break;
+      default:
+        break;
+    }
+    return parsedDay;
+  }
+
+  renderSwitch(frequency) {
+    const { occurence } = this.state;
+
+    switch (frequency) {
+      case 'yearly':
+        return (
+          <div>
+            On what day do you want to be reminded?
+            <ReactDayPicker handleUpdate={this.handleDays} resetDays="yearly" />
+          </div>
+        );
+      case 'monthly':
+        return (
+          <div>
+             <label htmlFor="occurence">
+                Every (?) Months:
+                <input type="number" id="occurence" value={occurence} onChange={(event) => this.handleChange(event, 'occurence')} min="0" max="12" required />
+              </label>
+              <div>On What day(s) do you want to be reminded?</div>
+              <ReactDayPicker handleUpdate={this.handleDays} resetDays="monthly" />
+          </div>
+        );
+      case 'weekly':
+        return (
+          <div>
+            <label htmlFor="occurence">
+              How many weeks in between do you want to be reminded?
+              <select id="occurence" placeholder="ex: Every 2 weeks" onChange={(event) => this.handleChange(event, 'occurence')} required >
+                <option value="1">Every Week</option>
+                <option value="2">Every 2 Weeks</option>
+                <option value="3">Every 3 Weeks</option>
+                <option value="4">Every 4 Weeks</option>
+              </select>
+            </label>
+            <div>Day of the week:</div>
+            <ReactDayPicker disabledDays={{ after: new Date(2022, 0, 8), before: new Date(2022, 0, 2) }} handleUpdate={this.handleDays} resetDays="weekly" />
```

Comment:
these dates seem hard-coded to the first week of january? maybe instead just grab today, then determine the `before` and `after` date bounds based off of that

Reply:
My thought process:

- We want to display 7 days for the user to choose, so then we can give them a reminder based on the day of the week. ex: [Sunday] || [Monday, Thursday]
- The change I'm pushing up will be to choose the first day of the month the user is in. Then allow them to choose a day from the 1st to the 7th. 
- My problem with choosing today, and then determining 7 days after, is that if it's towards the end of the month, the 7 days will leak into the next month, leading to user having to click the next month.
- Should we scrap the calendar on this weekly part of the form?